### PR TITLE
Update `Certificate` model

### PR DESCRIPTION
**Pull request by ammar92**
## Changes
Updated `Certificate` (and `Website`) model. Implemented a bit plugin for certificate expiry. 

## Issue ticket number and link
- https://github.com/minvws/nl-kat-coordination/issues/86

## Proof
_Please paste some screenshots or other proof of your (working) change here. If you feel that this is not required (e.g. this PR is trivial), note that here._

## Extra instructions for others
_This section may be skipped or omitted. Uncomment and answer the below questions if relevant._

- Does this PR introduce or depend on API-incompatible changes? If yes: what do other users/developers need to do or confirm before merging?
  - Yes, it contains modified models
  - Depends on https://github.com/minvws/nl-kat-rocky/pull/88

## Checklist for author(s):
- [x] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [ ] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [ ] If a non-trivial PR: This PR is properly linked to the project board (either directly or via an issue);
- [ ] If a non-trivial PR: I have added screenshots or some other proof that my code does what it is supposed to do;
- [x] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [ ] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [x] I have performed a self-review of my own code;
- [x] I have commented my code, particularly in hard-to-understand areas;
- [ ] I have made corresponding changes to the documentation, if necessary;
- [ ] I have written unit, integration, and end-to-end tests for the change that I made;

```
## Checklist for functional reviewer(s):
- [ ] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [ ] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [ ] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._
```

```
## Checklist for code reviewer(s):
- [ ] The code passes the CI tests and linters;
- [ ] The code does not bypass authentication or security mechanisms;
- [ ] The code does not introduce any dependency on a library that has not been properly vetted;
- [ ] The code does not violate Model-View-Template and our other architectural principles;
- [ ] The code contains docstrings, comments, and documentation where needed;
- [ ] The code prioritizes readability over performance where appropriate;
- [ ] The code conforms to our agreed coding standards.
```

### CODE REVIEW

File: bits/expiring_certificate/expiring_certificate.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from octopoes.models import OOI
+from octopoes.models.ooi.certificate import Certificate
+from octopoes.models.ooi.findings import KATFindingType, Finding
+from octopoes.models.ooi.web import Website
+
+THRESHOLD = datetime.timedelta(weeks=2)
+
+
+def run(input_ooi: Certificate, additional_oois: List[Website]) -> Iterator[OOI]:
+    # applies only to certificates assigned to websitess
```
Comment (underdarknl) [inline bits/expiring_certificate/expiring_certificate.py:13]: It should apply to all objects, no reason to not include other certs. However, we should only produce a finding for certificates that we see in use somewhere.

File: bits/expiring_certificate/expiring_certificate.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import datetime
+from typing import Iterator, List
+
+from octopoes.models import OOI
+from octopoes.models.ooi.certificate import Certificate
+from octopoes.models.ooi.findings import KATFindingType, Finding
+from octopoes.models.ooi.web import Website
+
+THRESHOLD = datetime.timedelta(weeks=2)
```
Comment (underdarknl) [inline bits/expiring_certificate/expiring_certificate.py:9]: Certificates have a UTC timestamp. We should make sure we check against UTC time and not local timezones.

File: bits/missing_certificate/missing_certificate.py [hunk 3]
```diff
@@ -1,28 +1,19 @@
...
 
 def run(
-    input_ooi: IPService,
-    additional_oois: List[Union[Certificate, Website]],
+    input_ooi: Website,
+    additional_oois,
 ) -> Iterator[OOI]:
 
-    if input_ooi.service.tokenized.name != "https":
+    if input_ooi.ip_service.tokenized.service.name.lower() != "https":
```
Comment (underdarknl) [inline bits/missing_certificate/missing_certificate.py:13]: The list of ip_services should be larger. imaps, pops etc all also use certs.
However, in the future the ip_service should contain a 'uses ssl/tls' flag which we fingerprint for.

**Review by underdarknl — COMMENTED** [review]
Some remarks, but its much nicer than the previous cert model.

File: bits/expiring_certificate/expiring_certificate.py (hunk 2 above)
Comment (ammar92) [reply bits/expiring_certificate/expiring_certificate.py:9]: Good point. We're not checking against local time zones, as long as plugins output such dates with a UTC timezone. The plugins related to this are updated to produce date times with UTC time zone.

File: bits/missing_certificate/missing_certificate.py (hunk 3 above)
Comment (ammar92) [reply bits/missing_certificate/missing_certificate.py:13]: That's true indeed. There are much more SANs to be implemented. For now, I've only implemented these few to keep the PR/ issue small. I'll create another ticket to implement more SANs and refer to this PR and issue.

File: bits/expiring_certificate/expiring_certificate.py (hunk 1 above)
Comment (ammar92) [reply bits/expiring_certificate/expiring_certificate.py]: Good point. See my other comment below. I rephrased this specific comment on line 13

**Review by underdarknl — APPROVED** [review]

**Pull request merged**